lib/features/children/parental_approval/widgets/parental_approval_dialog_content.dart

```diff
@@ -26,6 +26,9 @@ class ParentalApprovalDialogContent extends StatelessWidget {
   Future<void> _refreshHistory(BuildContext context) async {
     await context.read<FamilyOverviewCubit>().fetchFamilyProfiles();
     if (!context.mounted) return;
+    
+    await context.read<GoalTrackerCubit>().getGoal();
+    if (!context.mounted) return;
 
     // Exectue the following cubits in parallel
     await Future.wait([
```

Comment:
I would also remove getGoal from the "parallel" below, to not have it twice in this method

Reply:
Now you say this i'm actually wondering why that one doesnt work.. I'll do some more tests!

Reply:
also why  is `if (!context.mounted) return;` twice also

Reply:
https://dart.dev/tools/linter-rules/use_build_context_synchronously :)